#!/bin/sh /usr/share/dpatch/dpatch-run
## 04-gcc4.dpatch by Alexandre Francois <alexandre-francois@voila.fr>.
##
## DP: Fixes segfault with GCC 4.x (Closes: #330851).

@DPATCH@

diff -Naur pxe-1.4.2.orig/pxe.cc pxe-1.4.2/pxe.cc
--- pxe-1.4.2.orig/pxe.cc	2006-08-15 10:49:17.000000000 +0000
+++ pxe-1.4.2/pxe.cc	2006-08-15 10:48:33.000000000 +0000
@@ -263,7 +263,7 @@
 	if(-1 == chk)
 	{
 		std::cerr << "PXE daemon already running\n";
-		return(-1);
+		exit(-1);
 	}
 
 	// set the UID/GID to a low user
@@ -333,5 +333,5 @@
 		StartPxeService(configfile);
 	}
 	
-	return(0);
+	exit(0);
 }
